DataStructure/src/main/list/ResizableArray.java

```diff
@@ -165,4 +191,142 @@ public boolean isEmpty(){
     public int size(){
         return writeIndex;
     }
+
+    /**
+     * add all the elements from given resizableArray
+     * @param resizableArray
+     */
+    public void addAll(ResizableArray resizableArray){
+        if (list.length - size() < resizableArray.size()){
+            list = Arrays.copyOf(list,size()+resizableArray.size());
+        }
+        for (int i = 0; i < resizableArray.size(); i++){
+            add(resizableArray.get(i));
+        }
```

Comment:
System Array Copy method would be more efficient then calling add multiple times.